File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    q = deque(enumerate(priorities))
    while q:
        idx,p = q.popleft()
        
        if any( pp > p for _,pp in q):
            q.append((idx,p))
        else:
            answer+=1
            if(idx == location):
                return answer
        
        
    
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        Queue<int[]> q = new LinkedList<>();
        int n = priorities.length;
        for(int i=0; i<n;i++){
            q.offer(new int[]{i, priorities[i]});
        }
        
        while(!q.isEmpty()){
            
            int[] current = q.poll();
            boolean flag = false;
            for(int[] job: q){
                if (job[1] > current[1]){
                    flag = true;
                }
            }
            
            if(flag){
                q.offer(current);
            }else{
                answer++;
                if(current[0]==location){
                    break;
                }
            }
            
            
        }
        
        
        return answer;
    }
}